sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/sql/adapter/ModelFieldsAdapters.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql.adapter;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.sdk.nexmark.model.Auction;
+import org.apache.beam.sdk.nexmark.model.Bid;
+import org.apache.beam.sdk.nexmark.model.Person;
+
+/**
+ * Maps Java model classes to Beam SQL record types.
+ */
+public class ModelFieldsAdapters {
```

Comment:
<!--thread_id:cc_151126021_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:1-->
<!--section:context-quote-->
> **echauchot** wrote:
> Yes it makes a lot of sense, your aim is much clearer now. I agree with the low coupling and with the SRP too (even if there is not a lot of SRP in beam ;) ). Also, considering that the SQL converters might become useless in the future, extracting that conversion logic is a good thing indeed. The only conversion methods that belong to the model objects will be the encode/decode that are very unlikely to disappear in the future :). Can you just rename the names because they are so close to each others that it is a bit confusing.

<!--section:body-->
Renamed